ihmc-robotics-toolkit/src/test/java/us/ihmc/robotics/MatrixMissingToolsTest.java

```diff
@@ -313,4 +313,123 @@ public void testSetSelectedMatrixDiagonals()
+         MatrixTestTools.assertMatrixEquals(matrixDiagonalHand, matrixDiagonalResult, error);
+
+         // Test for skew matrix, simple 2x2 size of skew matrix
+         DMatrixRMaj skewMatrix = new DMatrixRMaj(2,2);
+         double skewTempValue = random.nextDouble();
+         skewMatrix.zero();
+         skewMatrix.set(0,1, skewTempValue);
+         skewMatrix.set(1,0,-skewTempValue);
+
+         DMatrixRMaj skewMatrixHand = new DMatrixRMaj(2,2);
+         DMatrixRMaj skewMatrixResult = new DMatrixRMaj(2,2);
+
+         //check square times
+         skewMatrixHand.zero();
+         skewMatrixHand.set(0,0, -skewTempValue*skewTempValue);
+         skewMatrixHand.set(1,1, -skewTempValue*skewTempValue);
+         MatrixMissingTools.power(skewMatrix,2,skewMatrixResult);
+         MatrixTestTools.assertMatrixEquals(skewMatrixHand, skewMatrixResult, error);
+
+         //check triple times
+         skewMatrixHand.zero();
+         skewMatrixHand.set(0,1, -skewTempValue*skewTempValue*skewTempValue);
+         skewMatrixHand.set(1,0, skewTempValue*skewTempValue*skewTempValue);
+         MatrixMissingTools.power(skewMatrix,3,skewMatrixResult);
+         MatrixTestTools.assertMatrixEquals(skewMatrixHand, skewMatrixResult, error);
+
+         //Check quadruple time
+         skewMatrixHand.zero();
+         skewMatrixHand.set(0,0, skewTempValue*skewTempValue*skewTempValue*skewTempValue);
+         skewMatrixHand.set(1,1, skewTempValue*skewTempValue*skewTempValue*skewTempValue);
+         MatrixMissingTools.power(skewMatrix,4,skewMatrixResult);
+         MatrixTestTools.assertMatrixEquals(skewMatrixHand, skewMatrixResult, error);
+
+
+
+         //         MatrixTestTools.assertMatrixEquals(matrixPowerTest, matrixCompareResult, error);
+      }
+   }
+
+   @Test
+   public void testElementWiseLessThan()
+   {
+      Random random = new Random(41584L);
+
+      int iters = 100;
+
+      for (int i = 0; i < iters; i++)
+      {
+         int rowSize = random.nextInt(5, 10);
+         int columnSize = random.nextInt(5, 10);
+
+         DMatrixRMaj matrixLess = RandomMatrices_DDRM.rectangle(rowSize, columnSize, random);
+         DMatrixRMaj matrixMore = new DMatrixRMaj(rowSize, columnSize);
+         for (int j = 0; j < rowSize; j++)
+         {
+            for (int k = 0; k < columnSize; k++)
+            {
+               matrixMore.set(j, k, matrixLess.get(j, k) + random.nextDouble(0.01, 0.05));
+            }
+         }
+
+         assertTrue(MatrixMissingTools.elementWiseLessThan(matrixLess, matrixMore));
+         assertFalse(MatrixMissingTools.elementWiseLessThan(matrixMore, matrixLess));
+
+         for (int j = 0; j < matrixLess.getNumElements(); j++)
+         {
+            DMatrixRMaj matrixLessNotAll = new DMatrixRMaj(matrixLess);
+            matrixLessNotAll.set(j, matrixMore.get(j) + 0.001);
+            assertFalse(MatrixMissingTools.elementWiseLessThan(matrixLess, matrixMore));
```

Comment:
it looks like you didn't use the new `matrixLessNotAll` object

Reply:
OH, you are right.

Reply:
Fixed